File: lista1/z3/zad3.py
import sys

from agent import *

tabu_size = int(1e4)
neighbourhood_size = int(1e3)


def read_values(file):
    with open(file) as f:
        matrix = []
        t, n, m = list(map(int, f.readline().split(' ')))
        for i in range(n):
            matrix.append(list(map(int, [c for c in f.readline().strip()])))
    return t, n, m, matrix


def find_path(t, n, m, g):
    grid = Grid(tabu_size, neighbourhood_size, n, m, g)
    solution, cost = grid.run_tabu(t)
    return solution, cost


def main():
    if len(sys.argv) != 4:
        raise Exception('Incorrect number of parameters')
    file_in, file_out, file_err = sys.argv[1], sys.argv[2], sys.argv[3]
    t, n, m, g = read_values(file_in)
    solution, cost = find_path(t, n, m, g)
    with open(file_out, 'w') as f:
        f.write(str(cost))
    with open(file_err, 'w') as f:
        f.write(''.join(solution))


if __name__ == '__main__':
    main()

File: lista3/z3/zad3.jl
include("./agent.jl")
using .Agent

function read_args()
    args = map(string, ARGS)
    file_in = args[1]
    file_out = args[2]
    file_err = args[3]
    return file_in, file_out, file_err
end

function read_grid(file_in)
    f_line, lines = nothing, nothing
    solutions = []
    open(file_in) do file
        f_line = split(strip(readline(file)), " ")
        lines = readlines(file)
    end
    t, n, m, s, p = map(x -> parse(Int, x), f_line)
    grid = Array{Int}(undef, n, m)
    for i = 1:n
        line = strip(lines[i])
        for j = 1:m
            grid[i,j] = parse(Int, line[j])
        end
    end
    for i = (n+1):(n+s)
        line = strip(lines[i])
        push!(solutions, map(x -> first(x), split(line, "")))
    end
    return t, n, m, s, p, grid, solutions
end

function write_result(file_out, file_err, steps, solution)
    open(file_out, "w") do file
        write(file, "$(steps)\n")
    end
    solution = string(solution...)
    open(file_err, "w") do file
        write(file, "$(solution)\n")
    end
end

function main()
    file_in, file_out, file_err = read_args()
    t, n, m, s, p, grid, solutions = read_grid(file_in)
    cost, solution = run_ga(t, n, m, s, p, grid, solutions, [])
    write_result(file_out, file_err, cost, solution)
end

main()
